pkg/services/object/acl/eacl/v2/xheader.go

```diff
@@ -21,16 +21,21 @@ type responseXHeaderSource struct {
 
 func (s requestXHeaderSource) GetXHeaders() []eaclSDK.Header {
 	ln := 0
+	xHdrs := make([][]session.XHeader, 0)
 
 	for meta := s.req.GetMetaHeader(); meta != nil; meta = meta.GetOrigin() {
-		ln += len(meta.GetXHeaders())
+		x := meta.GetXHeaders()
+
+		ln += len(x)
+
+		xHdrs = append(xHdrs, x)
 	}
 
 	res := make([]eaclSDK.Header, 0, ln)
-	for meta := s.req.GetMetaHeader(); meta != nil; meta = meta.GetOrigin() {
-		x := meta.GetXHeaders()
-		for i := range x {
-			res = append(res, sessionSDK.NewXHeaderFromV2(&x[i]))
+
+	for i := range xHdrs {
+		for j := range xHdrs[i] {
+			res = append(res, (xHeader)(xHdrs[i][j]))
```

Comment:
Why don't just replace a conversion? `sessionSDK.NewHeaderFromV2 -> x[i]`

Reply:
Replace with what? We need `eaclSDK.Header` interface, `xHeader` provides it.

Reply:
I mean `xHdrs` slice was not here before, this is probably a rebase artifact.

Reply:
Oh, got it, fixed.